solutions/standard/variables.tf

```diff
@@ -351,12 +351,18 @@ variable "existing_code_engine_project_id" {
 
 variable "enable_kibana_dashboard" {
   type        = bool
-  description = "Set it true to deploy Kibana in code engine. NOTE: Kibana image is coming direcly from the official registry (https://www.docker.elastic.co/) and not certified by the IBM."
+  description = "Set to true to deploy Kibana in Code Engine. NOTE: By default, the Kibana image will be pulled from the official Elastic registry (docker.elastic.co) and is not certified by IBM, however this can be overridden using the `kibana_image_reference` input."
   default     = false
 }
 
 variable "elasticsearch_full_version" {
```

Comment:
I really am not a fan of this variable name. It should be consistent with the `kibana_image_reference` input, and name `kibana_image_version` - thoughts @Ak-sky ?

Reply:
Right @ocofaigh, as this is specific to Kibana, and a user may confuse it with `elasticsearch_version﻿` variable- 
https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/blob/15af9b452ed795fd0a6b30e0a4cd1944d414266b/solutions/standard/variables.tf#L49
@aatreyee2506, please rename `elasticsearch_full_version`  to  `kibana_elasticsearch_full_version` and add this URL `https://github.com/elastic/kibana?tab=readme-ov-file#version-compatibility-with-elasticsearch` to the variable description for the user as a reference.

Reply:
We can use `kibana_image_version`, but below are the reasons why I added-

`_full_version`-  becuse we need `major.minor.patch` version number and not just `major.minor`. Kibana and Elasticsearch versions must match exactly for major, minor, and patch versions to avoid runtime errors.

`elasticsearch`- beacuse to get the full version of the deployed database (elastic search) which is required to deploy Kibana.

Here is a [reference](https://github.com/IBM/elasticsearch-kibana-codeengine/blob/bed82564ae14ecfeb979a469f3877322f84b958b/terraform/elastic.tf#L1) which says- 
```
# The data object below calls the ES URL in order to establish the full version of the deployed database
# because that is needed to deploy Kibana.
# The full version gets stored in a local variable es-full-version and then used in the codengine resources
```

Reply:
@Ak-sky But kibana don't release image tags with `major.minor` format. All their tags are `major.minor.patch` format:
![image](https://github.com/user-attachments/assets/def669f2-4abf-4982-939a-3aa3a40a1f8f)

Also its actually a better practise to lock into a digest instead of a tag version (as they may not be static), so we need to support that too.

So if we breakdown what an image reference looks like with correct terminology, it looks like this:
`[registry-url]/[namespace]/[image][:tag|@digest]`. So for kibana, that would be:

- registry-url: `docker.elastic.co`
- namespace: `kibana`
- image: `kibana`
- tag: `:8.15.4`
OR
- digest: `@sha256:0ba5cab0b781c3b2a242ef8b5f2544638fa9741fad03c218c58798c06c2bb175`

With all this in mind, I think we should have 2 variables:

- `kibana_registry_namespace_image` with a default value of `docker.elastic.co/kibana/kibana`
- `kibana_image_digest` with a default value of `null`. The variable description should day this:

> By default, when `enable_kibana_dashboard` is set to true, the DA will deploy a kibana using an image tag version that is applicable with the version of Elasticsearch deployed. Alternatively you can override this by supplying an image digest in the format of `sha256:xxxxx...`, however if doing so ensure the digest entered is an applicable version for your Elasticsearch instance.

Reply:
This is better and even if the user is going to use private registery, they can replace `kibana_registry_namespace_image` value.

But from security review aspect we will ask the user to input the registry url as you suggested? then no need to set the default values?

Reply:
lets go with having a default value, and then ask the security team to review. If needed, we can remove it in a new PR

Reply:
I'm also just thinking that we need to provide the ability to pass credentials incase the image registry the user picks requires authentication for pulling the kibana image. I created a follow up issue to track that: https://github.com/terraform-ibm-modules/terraform-ibm-icd-elasticsearch/issues/370

Reply:
I have made the changes.